From 1bb88d69e0ad951f9436a79e5a5d4bb35bf4076d Mon Sep 17 00:00:00 2001
From: Kevin Wolf <kwolf@redhat.com>
Date: Wed, 6 Nov 2013 14:41:15 +0100
Subject: [PATCH 39/81] apic: Document why cannot_instantiate_with_device_add_yet

RH-Author: Kevin Wolf <kwolf@redhat.com>
Message-id: <1383748882-22831-11-git-send-email-kwolf@redhat.com>
Patchwork-id: 55536
O-Subject: [RHEL-7.0 qemu-kvm PATCH v2 10/17] apic: Document why cannot_instantiate_with_device_add_yet
Bugzilla: 1001216
RH-Acked-by: Marcel Apfelbaum <marcel.a@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Reviewed-by: Peter Maydell <peter.maydell@linaro.org>
(cherry picked from pending upstream submission)
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
---
 hw/intc/apic_common.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/intc/apic_common.c |    6 +++++-
 1 files changed, 5 insertions(+), 1 deletions(-)

diff --git a/hw/intc/apic_common.c b/hw/intc/apic_common.c
index 208e7b1..ff413de 100644
--- a/hw/intc/apic_common.c
+++ b/hw/intc/apic_common.c
@@ -386,9 +386,13 @@ static void apic_common_class_init(ObjectClass *klass, void *data)
 
     dc->vmsd = &vmstate_apic_common;
     dc->reset = apic_reset_common;
-    dc->cannot_instantiate_with_device_add_yet = true; /* FIXME explain why */
     dc->props = apic_properties_common;
     idc->init = apic_init_common;
+    /*
+     * Reason: APIC and CPU need to be wired up by
+     * x86_cpu_apic_create()
+     */
+    dc->cannot_instantiate_with_device_add_yet = true;
 }
 
 static const TypeInfo apic_common_type = {
-- 
1.7.1

